scripts/aznfs_install.sh

```diff
@@ -183,73 +292,184 @@ esac
-            exit 0
+        # Check if the service name is "auto-update"
+        if [ "$SERVICE_NAME" == "auto-update" ]; then
+            # Compare the current version with the latest release
+            result=$(compare_versions "$current_version" "$RELEASE_NUMBER")
+            
+            # Check if an update is available
+            if [ "$result" -eq "1" ]; then
+                # Check if the user wants to perform the update
+                if [ "$user_wants_update" = "true" ]; then
+                    AZNFS_RELEASE="aznfs-${RELEASE_NUMBER}-1"
+                    # Create a flag file to indicate that an update is in progress
+                    touch /tmp/update_in_progress_from_watchdog.flag
+                else
+                    wecho "Version $RELEASE_NUMBER of AZNFS is available. Update to AZNFS $RELEASE_NUMBER for the latest features and improvements."
+                    wecho "Set AUTO_UPDATE_AZNFS=true in $CONFIG_FILE to auto-update" 
+                fi
+            else
+                pecho "AZNFS version $current_version is up-to-date or newer"
+            fi
+            
+        elif [ "$SERVICE_NAME" == "manual-update" ]; then
+            # Check if the current version matches the desired release number
+            if [ "$current_version" == "$RELEASE_NUMBER" ]; then
+                secho "AZNFS version $current_version is already installed."
+                exit 0
+            fi
+            
+            # Ask the user if they want to install the desired release
+            read -n 1 -p "AZNFS version $current_version is already installed. Do you want to install version $RELEASE_NUMBER? [Y/n] " result < /dev/tty
+            echo
+            if [ -n "$result" -a "$result" != "y" -a "$result" != "Y" ]; then
+                eecho "Installation aborted!"
+                exit 1
+            fi
         fi
-        read -n 1 -p "AZNFS version $current_version is already installed. Do you want to install version $RELEASE_NUMBER? [Y/n] " result < /dev/tty
-        echo
-        if [ -n "$result" -a "$result" != "y" -a "$result" != "Y" ]; then
-            eecho "Installation aborted!"
-            exit 1
+    fi
+        
+    # For watchdog, the flag file will always be present; otherwise, the service will be "manual-update"
+    if [ -f /tmp/update_in_progress_from_watchdog.flag ] || [ "$SERVICE_NAME" == "manual-update" ]; then
+        wget "https://github.com/Azure/AZNFS-mount/releases/download/${RELEASE_NUMBER}/${AZNFS_RELEASE}_amd64.deb" -P /tmp
+        apt install -y "/tmp/${AZNFS_RELEASE}_amd64.deb"
+        install_error=$?
+        rm -f "/tmp/${AZNFS_RELEASE}_amd64.deb"
+
+        if [ "$SERVICE_NAME" == "auto-update" ] && [ "$install_error" -eq 0 ]; then
+            pecho "AZNFS updates installed. Restarting aznfswatchdog to apply changes!"
+            systemctl daemon-reload
+            systemctl restart aznfswatchdog
         fi
```

Comment:
can't this be refactored into a function instead of repeating the same for both distros